src/randomx.cpp

```diff
@@ -93,7 +93,12 @@ extern "C" {
 	void randomx_init_cache(randomx_cache *cache, const void *key, size_t keySize) {
 		assert(cache != nullptr);
 		assert(keySize == 0 || key != nullptr);
-		cache->initialize(cache, key, keySize);
+		std::string data;
+		data.assign((const char *)key, keySize);
+		if (cache->cacheData != data) {
+			cache->initialize(cache, key, keySize);
+			cache->cacheData = data;
+		}
```

Comment:
This will leave the cache uninitialized if the key is empty (which is a valid key value). You should add `|| !cache->isInitialized()`.

Reply:
added isInit test